Automattic/jetpack
# Protect: Add existing plan/license key CTA

#### Description
As per the provided designs, adds a link to the header on the interstitial page so users with an existing plan or license key can bypass the upgrade prompt

#### Changes proposed in this Pull Request:

* Adds a CTA button for users who have an existing plan or license key that:
   * Initiates the checkout flow and:
      * Check for the presence of a scan-level license:
         * If found, redirects to the Protect plugin admin page
         * If not found, checks for any applicable unattached licenses and redirects to the license activation page in My Jetpack
         * If neither is found, redirects to the checkout with a scan plan
* Adds a plan check and status refresh after site registration is complete for the existing start-for-free button
* Adds state handling for the interstitial page button `isLoading` properties so each is provided a unique variable and treated independently

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203480626434857

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:





Existing plan flow (with or without an existing unattached license)
- Disconnect your site if it's connected.
- Make sure you have an attached plan in https://wordpress.com/me/purchases
- Go to the Jetpack Protect dashboard, and click the new "Click here to get started'" button in the header to initialize the connection flow.
- Confirm that after clicking "Approve", you get redirected to Protect admin dashboard, confirm you're connected (both blog and user token are saved) and you are presented the paid UI.

Existing license flow (without an existing plan)
- Go to https://cloud.jetpack.com/features/comparison and purchase the Security license. Do not attach it to a site.
- Disconnect your site if it's connected.
- Go to the Jetpack Protect dashboard and click the new "Click here to get started'" button in the header to initialize the connection flow.
- Click "Approve", you should get redirected to "My Jetpack -> Add a License Key" page.
- Confirm that you see your unattached license in the dropdown.

Extra credits:
- Ensure that if neither is applicable you are appropriately redirected to the plans page after authorizing a user connection
- When you have an existing plan, verify that clicking Start for Free from the Protect interstitial page loads the correct UI after registering a site-only connection.
